EIPS/eip-2386.md

```diff
@@ -0,0 +1,208 @@
+
+The `crypto` section follows the definition described in [EIP-2333](https://eips.ethereum.org/EIPS/eip-2333).
+
+This element MAY be present if the wallet type requires it.
+
+### Next Account
+
+The `nextaccount` provided is the next account to generate for wallets that require this information.  For example, `hierarchical deterministic` wallets create private keys based on a path that has an incrementing value (the first wallet will be `m/12381/60/0/0`, the second `m/12381/60/1/0`, the third `m/12381/60/2/0` etc.).
+
+This element MAY be present if the wallet type requires it.  If present it MUST be a non-negative integer.
+
+## JSON schema
+
+The walletstore follows a similar format to that of the keystore described in [EIP-2333](https://eips.ethereum.org/EIPS/eip-2333).
+
+```json
+{
+    "$ref": "#/definitions/Walletstore",
+    "definitions": {
+        "Walletstore": {
+            "type": "object",
+            "properties": {
+                "crypto": {
+                    "type": "object",
+                    "properties": {
+                        "kdf": {
+                            "$ref": "#/definitions/Module"
+                        },
+                        "checksum": {
+                            "$ref": "#/definitions/Module"
+                        },
+                        "cipher": {
+                            "$ref": "#/definitions/Module"
+                        }
+                    }
+                },
+                "name": {
+                    "type": "string"
+                },
+                "nextaccount": {
+                    "type": "integer"
+                },
+                "type": {
+                    "type": "string"
+                },
+                "uuid": {
+                    "type": "string",
+                    "format": "uuid"
+                },
+                "version": {
+                    "type": "integer"
+                }
+            },
+            "required": [
+                "name",
+                "type",
+                "uuid",
+                "version"
+            ],
+            "title": "Walletstore"
+        },
+        "Module": {
+            "type": "object",
+            "properties": {
+                "function": {
+                    "type": "string"
+                },
+                "params": {
+                    "type": "object"
+                },
+                "message": {
+                    "type": "string"
+                }
+            },
+            "required": [
+                "function",
+                "message",
+                "params"
+            ]
+        }
+    }
+}
+```
+
+## Rationale
+
+A standard for walletstores, similar to that for keystores, provides a higher level of compatibility between wallets and allows for simpler wallet and key interchange between them.
+
+## Test Cases
+
+### Non-deterministic Test Vector
```

Comment:
How is a key stored in the non-hd case?

Reply:
There is no key for non-deterministic wallets as the keys are created from random data.

Reply:
What is the point supporting this if no key is stored? It seems to be a non very useful feature. I may be just confused by the intent.

Reply:
The type of the wallet defines if it is deterministic or not.  If deterministic it needs a seed/key, if not it doesn't.  So the value is required for some wallets and not others.  The standard should allow for both options.

Reply:
Is there any progress on this being merged?

If there is confusion about the purpose it may help to look at https://github.com/wealdtech/go-eth2-wallet#usage regarding the purpose of walletstore.  Walletstore is a container that defines how keys are generated within a particular wallet, and provides a container for multiple keys (shown as "account"s) in the diagram at that URL).

Reply:
I am really confused of what use the non-deterministic vector has.

The EIP/ERC should be self contained so that its abstraction and motivation clearly explains everything, without the need for looking for external references.

Reply:
I've updated the motivation section, which should provide more clarity as to their purpose.

Reply:
Did the update help clarify the purpose of this EIP?  If not, please let me know what is unclear and I will attempt to provider further detail

Reply:
Please could someone let me know what to do to move this along?

Reply:
I'm also not sure if the non-HD wallet carries it's weight. From what I can tell there's nothing that links the non-HD wallet to generated keys/keystores so it might as well not exist.

Maybe you could help me understand by explaining why anyone would want to export a non-HD wallet from one machine to another?

The non-HD wallet option makes the JSON structure non-trivially more complex; the `crypto` and `path` keys might or might-not exist based upon the value of `type`. Unless I'm missing the purpose of it, it would be a shame to say that one can't be compatible with EIP-2386 unless they implement something that has no practical cryptographic purpose.